src/fairseq2/recipes/wav2vec2/_train.py

```diff
@@ -241,45 +247,57 @@ def load_wav2vec2_trainer(
         batch_shuffle_window=config.dataset.batch_shuffle_window,
         num_accumulate=config.trainer.gradient_accumulation,
         num_prefetch=config.dataset.num_prefetch,
+        example_shuffle_window=config.dataset.example_shuffle_window,
+        use_fbank=config.dataset.use_fbank,
+        no_padding=config.dataset.no_padding,
+        npc=config.dataset.npc,
         seed=seed,
         extras=config.dataset.extras,
     )
 
     data_reader = dataset.create_reader(
         config.dataset.train_split,
         gangs.dp,
-        config.dataset.min_audio_len,
-        config.dataset.max_audio_len,
-        read_options,
+        min_audio_len=config.dataset.min_audio_len,
+        max_audio_len=config.dataset.max_audio_len,
+        options=read_options,
     )
 
     seed += 1
 
     # Initialize the validation unit.
     if config.dataset.valid_split is not None:
-        valid_unit = Wav2Vec2EvalUnit(criterion, gangs)
 
+        config.dataset.extras["is_binarized"] = False
         read_options = SpeechReadOptions(
             batching=batching,
             dtype=config.trainer.dtype,
             normalize_audio=config.dataset.normalize_audio,
             sync_mode=SyncMode.UNTIL_LAST,
             num_prefetch=config.dataset.num_prefetch,
+            example_shuffle_window=config.dataset.example_shuffle_window,
+            use_fbank=config.dataset.use_fbank,
+            no_padding=config.dataset.no_padding,
+            npc=config.dataset.npc,
             seed=seed,
             extras=config.dataset.extras,
         )
 
-        valid_data_reader = dataset.create_reader(
-            config.dataset.valid_split,
-            gangs.dp,
-            config.dataset.min_audio_len,
-            config.dataset.max_audio_len,
-            read_options,
-        )
-
-        valid_units = [valid_unit]
-
-        valid_data_readers = [valid_data_reader]
+        valid_units = []
+        valid_data_readers = []
+        valid_splits = (config.dataset.valid_split).split(",")
```

Comment:
add a comment here about the format

Reply:
Done at the configuration level and added examples to the yaml. https://github.com/facebookresearch/fairseq2/blob/wav2vec2_trainer/recipes/wav2vec2/ssl/configs/base_960h.yaml